### DISCUSSION TITLE
How to swap accounts?

### DISCUSSION THREAD

**Discussion by DerekPeng303** [PrismarineJS/mineflayer#2392] [Q&A] [answered]
So I have 2 Minecraft accounts that I would like to swap between when using Mineflayer. However, when I used Microsoft to authenticate (where it sent the code and the link), I realised I couldn't change which account I was using. Even if I set the username and password for the alt, it would still use the other account to log in. 

For example:
var bot = mineflayer.createBot({
    username: '123@gmail.com',
    password: 'password1',
    auth: 'microsoft', 
    host: 'serverip',
    version: '1.8.9'
})
This would still use the account that had the email abc@gmail.com and not the one with 123@gmail.com

Any ideas on how to change which account is used?

Edit:
For some reason, it randomly changes which account it uses. No idea why...

**Comment by DerekPeng303** [comment, marked as answer]
I found out that deleting the nmp-cache in the .minecraft folder allows you to re-log in.

**Reply by IceTank** [reply]
When using microsoft you do not need the password field. I have not been able to reproduce this issue when using different microsoft accounts. Maybe try it without the password field. The login code should check if the provided username/email is different to the username or email stored in the nmp cache.

**Reply by DerekPeng303** [reply]
Thanks for the reply! I've tried it without the password field and it still does not change the account it is using. For the login code checking the provided email, both login details are stored which might mess with the checking and thus stop it from changing accounts? Honestly, deleting the nmp-cache isn't too annoying, so you don't need to waste your time debugging for me.
